src/pages/_app.js

```diff
@@ -38,22 +34,30 @@ export default function App({ Component, pageProps }) {
     setTransactions([newTransaction, ...transactions]);
   }
 
+  function handleEditTransaction(id, updatedTransaction) {
+    setTransactions(
+      transactions.map((transaction) =>
+        transaction.id === id ? { ...transaction, ...updatedTransaction } : transaction),
+    );
+    handleAlert("Transaction successfully updated!");
+  }
+
   function handleDelete(id) {
-    const updatedData = transactions.filter((item) => item.id !== id);
-    setTransactions(updatedData);
+    setTransactions(transactions.filter((item) => item.id !== id));
```

Comment:
I am right that ```item```is a transaction? For better readability I would recommend to keep the same names for equal items

Reply:
yep, changed accordingly